# LinearRegression 

## Example

In [10]:
from sklearn.linear_model import LinearRegression 
 
X = [[5],[9],[10],[3],[5],[7]] 
y = [6,11,6,4,6,9] 
regressor = LinearRegression() 
regressor.fit(X, y) 
print(regressor.predict([[0]])) 

[3.15492958]


# Ride sharing Linear regression

In [3]:
import pandas as pd
from sklearn.linear_model import LinearRegression
### Load the ride sharing dataset into a pandas dataframe
df = pd.read_csv(r'files\Rides.csv')
df.head()

Ride ID           City  Rider ID  Ride Time  Ride Fare Ride Date  Ride Hour
0        1       New York       100       15.0       10.0  1/1/2022        9.0
1        2  San Francisco       101       30.0       25.0  1/1/2022       10.0
2        3    Los Angeles       102       45.0       30.0  1/1/2022       11.0
3        4       New York       103       20.0       15.0  1/1/2022       12.0
4        5  San Francisco       104       10.0        7.0  1/2/2022        9.0

# Drop rows with missing values:

# Er mogen geen NaN waardes in staan anders doet het model het niet.

In [11]:
dfcleaned = df.dropna()

In [12]:
## Select the independent variables (ride_time and distance) and dependent variable (fare), respectively, from the dataframe "df".
X = dfcleaned[['Ride Time', 'Ride Hour']]
y = dfcleaned['Ride Fare']

### Fit a linear regression model to the data
reg = LinearRegression().fit(X, y)

### Print the coefficients of the linear regression model
print('Coefficients:', reg.coef_)

Coefficients: [ 0.74158    -0.19120212]


0.7 zie hierboven ligt bijna tegen 1 aan: betekent dat er een correlatie is tussen Ride Time en Ride Fare.
Dus als de tijd langer duurt, wordt de Fare ook langer.
-0.19 zit bijna tegen de 0 aan, Zero correlatie, dus betekent dat er geen relatie is tussen Ride Hour, dus uur van de dag waarop een rit plaats vind en de Ride Fare.
This code performs a linear regression on the ride sharing dataset using the scikit-learn library. Here's a breakdown of the code:

- The first line imports the pandas library, which is used for data manipulation and analysis.
- The second line imports the LinearRegression class from the scikit-learn library, which is used for modeling linear regression.
- The third line loads the ride sharing dataset into a pandas dataframe named "df".
- The fourth and fifth lines select the independent variables (ride_time and distance) and dependent variable (fare), respectively, from the dataframe "df".
- The sixth line fits a linear regression model to the data using the selected variables.
- The seventh line prints the coefficients of the linear regression model.

The coefficients represent the slopes of the linear regression line for each independent variable. 
In this case, there are two independent variables (Ride Time and Ride Hour) and one dependent variable (fare). 
The output of the code will be a list of two coefficients, one for each independent variable. 
These coefficients will indicate how much the fare changes for a one unit increase in each independent variable, holding all other variables constant.

These coefficients will indicate how much the fare changes for a one unit increase in each independent variable, holding all other variables constant.
That should mean that the first one: coefficient: 0.74 will indicate that lets say Ride time= 10 (minutes i guess) and Fare = 7 (Ride id 5) that

if Ride time which has a positive correlation is going from 10 minutes to 11 minutes (1 unit added),
the Fare will increase by 0.74 so the Fare will be Ride Fare 7 + 0.74 = 7.74